common/collections.ts

```diff
@@ -268,13 +285,34 @@ export class Bytes extends ByteArray {
     return changetype<Bytes>(uint8Array)
   }
 
+  /**
+   * Convert the string `hex` which must consist of an even number of
+   * hexadecimal digits to a `ByteArray`. The string `hex` can optionally
+   * start with '0x'
+   */
+  static fromHexString(str: string): Bytes {
```

Comment:
Thanks pt2! lol

It's kind of annoying we have to make duplicate methods for those so users don't need to do the upcasting.